collectfast/strategies/boto3.py

```diff
@@ -29,6 +49,51 @@ def _clean_hash(quoted_hash: Optional[str]) -> Optional[str]:
         assert quoted_hash[0] == quoted_hash[-1] == '"'
         return quoted_hash[1:-1]
 
+    def get_aws_hash(self, open_handle: IO) -> str:
+        """Calculate multipart-friendly hash using `multipart_chunksize` chunk size."""
+        chunk_hashes = []
+        while True:
+            data = open_handle.read(multipart_chunksize)
+            if not data:
+                break
+            chunk_hashes.append(hashlib.md5(data))
```

Comment:
What do you think about breaking this out into a helper function? That could look something like this:

```python
def read_chunks(stream: IO, chunksize: int) -> Iterable[str]:
    while True:
        data = stream.read(chunksize)
        if not data:
            break
        yield data
```

Then we'd gain the benefit of making `chunk_hashes` immutable:

```python
chunk_hashes = tuple(
    hashlib.md5(chunk) for chunk in read_chunks(open_handle, multipart_chunksize)
)

Reply:
I like it!